### PR TITLE
ESLint Cleanup

### PR DESCRIPTION
**Pull request by JackyChiu**
## ESLint
Task: #45 
ESLint is a linter for Javascript. It provides indications of bad practices and styling problems. With this addition we will have a more consistent code base that provides an easy way to check for errors in best practices. After an integration with Travis, PR's will automatically be checked with the lint as well. 

## How to use the lint?
```bash
npm install
npm run lint
# Quick fix for lint errors (such as tabbing, commas)
npm run lint:fix

# For React
npm run build
# ESlint will provide errors here
```

## Add ESLint to your editior
- Atom: Add a package for the linter
- Vim: Add Syntactic and copy Jacky's dotfiles (Ask me for help if needed)
By having the lint in your editor you won't have to run the command every time. Just before making a PR. 

## How to review this huge PR? 
I will point out areas of logic changes, you can also checkout to my branch and run the application
Also here are some common things that were changed:
- Errors should be thrown instead of string, Errors can be paused with Chrome DevTools and contain the call stack
- No consoles in production code, you can use the console while developing but writing automated test and using the debug script will serve you better (Finally we know how to debug) 
- Extra commas, yes. they provide clearer diffs 
- Two-space tabs, all the style guides do this so rip to 4 spaces. 
- Pure functions instead of classes for dumb components 
- PropTypes for components,  provides validation for props, and should be used to prevent invalid props  (Haven't been fully implemented) 

## In future
- ESLint is complaining about my code! Try solving your problem a different way, ESLint is only trying to enforce best practices
- Try not to disable rules when ESLint complains, insteead google the error and you'll probably learn something. Otherwise add the rules and in the PR we will decide if it's a valid reason 

## BTW
Currently there are still some errors, I will fix them, waiting on @kirinrastogi's tests before I refactor said function

### CODE REVIEW

File: scripts/fetchArticles.js [hunk 1]
```diff
@@ -2,75 +2,53 @@ const request = require('request-promise')
...
 const REFRESH = 3600000
 
-module.exports.startFetchCycle = () => {
-    setInterval(() => {
-        updateNews()
-        console.log('ran') 
-    }, REFRESH)
-}
-
-function getWords(article) {
```
Comment (JackyChiu) [inline scripts/fetchArticles.js]: Removed unused function

File: scripts/fetchArticles.js [hunk 2]
```diff
@@ -2,75 +2,53 @@ const request = require('request-promise')
...
+      headlines.forEach((old) => {
+        Article.findOne({ url: old.url })
+          .then((article) => {
+            if (article) return
+            const newArticle = new Article(old)
+            newArticle.keywords = getPhrases(article)
+            newArticle.votes = 0
+            newArticle.save()
+          })
+          .catch(err => err)
```
Comment (JackyChiu) [inline scripts/fetchArticles.js]: Ignore failing cases for saving new article, instead should be abstracted to another function for testability

File: scripts/fetchArticles.js [hunk 3]
```diff
@@ -2,75 +2,53 @@ const request = require('request-promise')
...
+      })
+    })
+    .catch(err => err)
```
Comment (JackyChiu) [inline scripts/fetchArticles.js]: TODO: Have the function actually return the err if fails

File: routes/article.js [hunk 4]
```diff
@@ -1,140 +1,132 @@
...
             })
-        })
+            keyword.save()
+          })
+      })
+      res.send(articleChanged)
+    })
+    .catch((err) => {
+      res.status(500).send({
+        err: err.toString(),
```
Comment (JackyChiu) [inline routes/article.js]: Changed errors to always respond as a string

**Comment by JackyChiu** [comment]
TODO: Change Heroku to use production env after this merges

File: routes/article.js [hunk 5]
```diff
@@ -1,140 +1,131 @@
...
-const Keyword = require('../models/KeywordModel.js')
+const Article = require('../models/ArticleModel')
+const Keyword = require('../models/KeywordModel')
 
 const router = express.Router()
 
+function getNextArticles(findParams, sortParams) {
+  return new Promise((resolve, reject) => {
+    Article.find(findParams)
+      .sort(sortParams)
```
Comment (JackyChiu) [inline routes/article.js]: Promise style change: Call a promise before actually promise chaining. (Article is a variable not a promise)

Comment (VikramBombhi) [inline routes/article.js]: Can u expand on this more? what benefits does this have? iv seen it alot and i agree that it is more intuitive and cleaner but is there anything else?

Comment (JackyChiu) [inline routes/article.js]: It's just more intuitive since there's nothing really to chain in the former, until a promise has been returned

Comment (VikramBombhi) [inline routes/article.js]: alright got it

File: scripts/fetchArticles.js (hunk 2 above)
Comment (VikramBombhi) [inline scripts/fetchArticles.js]: how long did u spend deciding on this? the struggle must have been real

Comment (JackyChiu) [inline scripts/fetchArticles.js]: @VikramBombhi A long time man, but I think the solution is good, but might require mocking a db

**Comment by VikramBombhi** [comment]
LGTM SHIP IT

File: routes/article.js (hunk 5 above)
Comment (kirinrastogi) [inline routes/article.js]: I like this change

**Comment by kirinrastogi** [comment]
LGTM

**Comment by JackyChiu** [comment]
Opps it was suppose to be task #41

**Comment by VikramBombhi** [comment]
just edit ur pr instead of leaving it as a comment

**Comment by JackyChiu** [comment]
@VikramBombhi If you edit the PR, people assigned to the PR won't get notified, so any edits would go unnoticed

**Pull request merged**